packages/next-codemod/transforms/next-lint-to-eslint-cli.ts

```diff
@@ -0,0 +1,822 @@
+
+    // Check if Next.js configs are already imported
+    const hasNextConfigs =
+      configContent.includes('next/core-web-vitals') ||
+      configContent.includes('next/typescript')
+
+    // TypeScript config files need special handling
+    if (
+      configPath.endsWith('.ts') ||
+      configPath.endsWith('.mts') ||
+      configPath.endsWith('.cts')
+    ) {
+      console.log('   TypeScript config files require manual migration')
+      console.log('   Please add the following to your config:')
+      console.log('   - Import: import { FlatCompat } from "@eslint/eslintrc"')
+      console.log(
+        '   - Extend: ...compat.extends("next/core-web-vitals"' +
+          (isTypeScript ? ', "next/typescript"' : '') +
+          ')'
+      )
+      return false
+    }
+
+    // Parse the file using jscodeshift
+    const j = createParserFromPath(configPath)
+    const root = j(configContent)
+
+    // Determine if it's CommonJS or ES modules
+    const isCommonJS =
+      configPath.endsWith('.cjs') ||
+      (configPath.endsWith('.js') &&
+        root
+          .find(j.AssignmentExpression, {
+            left: {
+              type: 'MemberExpression',
+              object: { name: 'module' },
+              property: { name: 'exports' },
+            },
+          })
+          .size() > 0)
+
+    // Find the exported array
+    let exportedArray = null
+    let exportNode = null
+
+    if (isCommonJS) {
+      // Look for module.exports = [...]
+      const moduleExports = root.find(j.AssignmentExpression, {
+        left: {
+          type: 'MemberExpression',
+          object: { name: 'module' },
+          property: { name: 'exports' },
+        },
+        right: { type: 'ArrayExpression' },
+      })
+
+      if (moduleExports.size() > 0) {
+        exportNode = moduleExports.at(0)
+        exportedArray = exportNode.get('right')
+      }
+    } else {
+      // Look for export default [...]
+      const defaultExports = root.find(j.ExportDefaultDeclaration, {
+        declaration: { type: 'ArrayExpression' },
+      })
+
+      if (defaultExports.size() > 0) {
+        exportNode = defaultExports.at(0)
+        exportedArray = exportNode.get('declaration')
+      } else {
+        // Look for const variable = [...]; export default variable
+        const defaultExportIdentifier = root.find(j.ExportDefaultDeclaration, {
+          declaration: { type: 'Identifier' },
+        })
+
+        if (defaultExportIdentifier.size() > 0) {
+          const varName = defaultExportIdentifier.at(0).get('declaration')
+            .value.name
+          const varDeclaration = root.find(j.VariableDeclarator, {
+            id: { name: varName },
+            init: { type: 'ArrayExpression' },
+          })
+
+          if (varDeclaration.size() > 0) {
+            exportedArray = varDeclaration.at(0).get('init')
+          }
+        }
+      }
+    }
+
+    if (!exportedArray) {
+      console.log(
```

Comment:
This will be easy to ignore by users, better to have a more explict warning here, e.g. use `picocolors` to output it as warning with console.error or console.warn